### PR TITLE
Resolve "Forgot password"

### LINKED ISSUE #23: Add forgot password
Pages:

- Simple page with 1 email input to enter `email`.
- Another simple page to enter `token`.

Endpoints:

- `auth/password/reset/`
- `auth/password/reset/confirm/`

### PR DESCRIPTION
**Pull request by AmerAnsari**
Closes #23

### CODE REVIEW

File: src/app/components/reset-password/reset-password.component.ts [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+  /**
+   * User change password form
+   */
+  resetPasswordForm: FormGroup;
+
+  /**
+   * Success change password message
+   */
+  // successPasswordMessage: string;
```
Comment (AmirSavand) [inline src/app/components/reset-password/reset-password.component.ts]: Remove this line.

File: src/app/components/reset-password/reset-password.component.ts [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+      new_password2: ['', Validators.required],
+      token: ['', Validators.required],
+    });
+  }
+
+  /**
+   * Send reset password token to user email
+   */
+  tokenSubmit(): void {
+    if (this.loading) {
```
Comment (AmirSavand) [inline src/app/components/reset-password/reset-password.component.ts]: Remove loading check, just disable the submit.

File: src/app/components/reset-password/reset-password.component.ts [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+      return;
+    }
+    this.loading = true;
+    /**
+     * API call
+     */
+    this.authService.resetPassword(this.tokenForm.get('email').value).subscribe((data: { detail: string }) => {
+      this.loading = false;
+      this.successTokenMessage = data.detail;
```
Comment (AmirSavand) [inline src/app/components/reset-password/reset-password.component.ts]: Just use a `boolean` and rename it to `success` just like `loading` that is for both forms.

File: src/app/components/reset-password/reset-password.component.ts [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+      this.successTokenMessage = '';
+      this.tokenError = error.error;
+    });
+  }
+
+  /**
+   * reset password confirm
+   */
+  resetPasswordConfirm(): void {
+    if (this.loading) {
```
Comment (AmirSavand) [inline src/app/components/reset-password/reset-password.component.ts]: Same as above.

File: src/app/components/reset-password/reset-password.component.ts [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+    /**
+     * API call
+     */
+    this.authService.resetPasswordConfirm({
+      new_password1: this.resetPasswordForm.get('new_password1').value,
+      new_password2: this.resetPasswordForm.get('new_password2').value,
+      uid,
+      token,
+    }).subscribe((data: { detail: string }): void => {
+      // this.successPasswordMessage = data.detail;
```
Comment (AmirSavand) [inline src/app/components/reset-password/reset-password.component.ts]: Remove comment line.

File: src/app/components/reset-password/reset-password.component.ts [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+      this.resetPasswordError = {};
+      this.resetPasswordForm.reset();
+      /**
+       * Navigate to sign in page
+       */
+      this.router.navigateByUrl('/sign-in');
+    }, (error: HttpErrorResponse): void => {
+      this.loading = false;
+      // this.successPasswordMessage = '';
```
Comment (AmirSavand) [inline src/app/components/reset-password/reset-password.component.ts]: Removed comment line.

File: src/app/components/sign-in/sign-in.component.html [hunk 7]
```diff
@@ -17,6 +17,8 @@
               <button class="btn btn-primary rounded btn-block" [disabled]="loading">Submit</button>
             </div>
           </form>
+          <!-- Reset password -->
+          <a class="small" routerLink="/reset-password">I forgot my password!</a>
```
Comment (AmirSavand) [inline src/app/components/sign-in/sign-in.component.html]: Change label to "Forgot password?".

File: src/app/services/auth.service.ts [hunk 8]
```diff
@@ -187,4 +187,29 @@ export class AuthService {
   }): Observable<{ detail: string }> {
     return this.http.post<{ detail: string }>(`${ApiService.BASE}auth/password/change/`, payload);
   }
+
+  // Reset user password
```
Comment (AmirSavand) [inline src/app/services/auth.service.ts]: Remove, unnecessary.

File: src/app/components/reset-password/reset-password.component.ts (hunk 1 above)
Comment (AmerAnsari) [reply src/app/components/reset-password/reset-password.component.ts]: Done

File: src/app/components/reset-password/reset-password.component.ts (hunk 2 above)
Comment (AmerAnsari) [reply src/app/components/reset-password/reset-password.component.ts]: Done

File: src/app/components/reset-password/reset-password.component.ts (hunk 3 above)
Comment (AmerAnsari) [reply src/app/components/reset-password/reset-password.component.ts]: No, we need the success message

File: src/app/components/reset-password/reset-password.component.ts (hunk 4 above)
Comment (AmerAnsari) [reply src/app/components/reset-password/reset-password.component.ts]: Done

File: src/app/components/reset-password/reset-password.component.ts (hunk 5 above)
Comment (AmerAnsari) [reply src/app/components/reset-password/reset-password.component.ts]: Done

File: src/app/components/reset-password/reset-password.component.ts (hunk 6 above)
Comment (AmerAnsari) [reply src/app/components/reset-password/reset-password.component.ts]: Done

File: src/app/components/sign-in/sign-in.component.html (hunk 7 above)
Comment (AmerAnsari) [reply src/app/components/sign-in/sign-in.component.html]: Done

File: src/app/services/auth.service.ts (hunk 8 above)
Comment (AmerAnsari) [reply src/app/services/auth.service.ts]: Done

File: src/app/components/reset-password/reset-password.component.ts (hunk 3 above)
Comment (AmerAnsari) [reply src/app/components/reset-password/reset-password.component.ts]: Done, rename to "success"

**Pull request merged**